src/clr-angular/data/datagrid/datagrid-action-overflow.ts

```diff
@@ -1,37 +1,63 @@
 /*
- * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, EventEmitter, Input, OnDestroy, Output } from '@angular/core';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  ElementRef,
+  NgZone,
+  Inject,
+  PLATFORM_ID,
+} from '@angular/core';
 
 import { Point } from '../../popover/common/popover';
 
 import { RowActionService } from './providers/row-action-service';
 import { ClrCommonStrings } from '../../utils/i18n/common-strings.interface';
+import { isPlatformBrowser } from '@angular/common';
+import { take, debounceTime } from 'rxjs/operators';
+
+let clrDgActionId = 0;
 
 @Component({
   selector: 'clr-dg-action-overflow',
   template: `
-        <button (click)="toggle($event)" type="button" class="datagrid-action-toggle" #anchor>
+        <button (click)="toggle($event)" type="button" class="datagrid-action-toggle" #anchor role="button" 
+                    aria-haspopup="true" [attr.aria-controls]="popoverId" [attr.aria-expanded]="open">
             <clr-icon shape="ellipsis-vertical" [attr.title]="commonStrings.rowActions"></clr-icon>
         </button>
         <ng-template [(clrPopoverOld)]="open" [clrPopoverOldAnchor]="anchor" [clrPopoverOldAnchorPoint]="anchorPoint"
                      [clrPopoverOldPopoverPoint]="popoverPoint">
-            <div #menu class="datagrid-action-overflow" (clrOutsideClick)="close($event)" [clrStrict]="true">
+            <div [attr.class]="overflowClassName" (clrOutsideClick)="close($event)" [clrStrict]="true" 
+                    role="menu" [attr.id]="popoverId" [attr.aria-hidden]="!open">
                 <ng-content></ng-content>
             </div>
         </ng-template>
     `,
 })
 export class ClrDatagridActionOverflow implements OnDestroy {
+  public overflowClassName = 'datagrid-action-overflow';
   public anchorPoint: Point = Point.RIGHT_CENTER;
   public popoverPoint: Point = Point.LEFT_CENTER;
 
-  constructor(private rowActionService: RowActionService, public commonStrings: ClrCommonStrings) {
+  constructor(
+    private rowActionService: RowActionService,
+    public commonStrings: ClrCommonStrings,
+    private ref: ElementRef,
```

Comment:
Would you mind renaming it as `elRef` or `elementRef`? we usually have prefix for element refs just to avoid confusions from ViewRef, ComponentRef, etc...

Reply:
Done!